diff -urNp -x '*.orig' libfreenect-0.6.2.org/OpenNI2-FreenectDriver/CMakeLists.txt libfreenect-0.6.2/OpenNI2-FreenectDriver/CMakeLists.txt
--- libfreenect-0.6.2.org/OpenNI2-FreenectDriver/CMakeLists.txt	2021-02-12 00:51:27.000000000 +0100
+++ libfreenect-0.6.2/OpenNI2-FreenectDriver/CMakeLists.txt	2021-03-07 21:32:13.949845940 +0100
@@ -24,7 +24,7 @@ include_directories(${PROJECT_SOURCE_DIR
 
 target_compile_features(FreenectDriver PUBLIC cxx_std_11 cxx_constexpr)
 
-target_link_libraries(FreenectDriver freenectstatic ${MATH_LIB})
+target_link_libraries(FreenectDriver freenectstatic ${MATH_LIB} -lpthread)
 
 install (TARGETS FreenectDriver
   DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}/OpenNI2-FreenectDriver")
