--- libfreenect-0.6.0/wrappers/python/CMakeLists.txt.orig	2019-10-30 01:30:13.000000000 +0100
+++ libfreenect-0.6.0/wrappers/python/CMakeLists.txt	2020-03-15 08:31:40.236407736 +0100
@@ -50,7 +50,7 @@
 # CMake populates this var automatically but we need to specify prefix
 execute_process(COMMAND
   ${Python${Python_BUILD_VERSION}_EXECUTABLE}
-    -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(plat_specific=False, standard_lib=False, prefix='${CMAKE_INSTALL_PREFIX}'))"
+    -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(plat_specific=True, standard_lib=False, prefix='${CMAKE_INSTALL_PREFIX}'))"
   OUTPUT_VARIABLE Python${Python_BUILD_VERSION}_SITELIB OUTPUT_STRIP_TRAILING_WHITESPACE)
 
 # Figure out numpy include path
--- libfreenect-0.6.0/cmake_modules/SetupDirectories.cmake.orig	2019-10-30 01:30:13.000000000 +0100
+++ libfreenect-0.6.0/cmake_modules/SetupDirectories.cmake	2020-03-15 08:31:53.559668891 +0100
@@ -8,7 +8,7 @@
 STRING (TOLOWER ${PROJECT_NAME} projectNameLower)
 SET (PROJECT_INCLUDE_INSTALL_DIR "include/${projectNameLower}")
 SET (PROJECT_MANPAGE_INSTALL_DIR "share/man")
-SET (PROJECT_LIBRARY_INSTALL_DIR "lib")
+SET (PROJECT_LIBRARY_INSTALL_DIR "lib${LIB_SUFFIX}")
 
 MESSAGE (STATUS "${PROJECT_NAME} will be installed to ${CMAKE_INSTALL_PREFIX}")
 MESSAGE (STATUS "Headers will be installed to ${CMAKE_INSTALL_PREFIX}/${PROJECT_INCLUDE_INSTALL_DIR}")
